File: ai-ml/lesson-06-linear-regression/hard/p01-solve.py
"""
LESSON — Linear Regression
HARD P01 — Build a complete regression pipeline: load a real dataset (e.g., California Housing from sklearn), preprocess (scale features), train, evaluate with MSE/R², and plot actual vs predicted values.
==================================================

CONCEPT:
  See concepts.md in this lesson folder for detailed explanations.

PROBLEM:
  Build a complete regression pipeline: load a real dataset (e.g., California Housing from sklearn), preprocess (scale features), train, evaluate with MSE/R², and plot actual vs predicted values


Write a function `solve()` that implements the solution.

# Check your answer: compare with solutions/p01-solution.py
"""

TRY THIS INPUT:
  Add this test code at the bottom of your file:

  ```python
  data = fetch_california_housing()
  X, y = data.data, data.target
  feature_names = data.feature_names

  # Call your function
  result = solve(...)
  print(result)
  ```

EXPECTED OUTPUT:
  Run the solution file to see expected output:
    python3 hard/solutions/p01-solution.py

  Then compare your output format with theirs.

# === WRITE YOUR CODE BELOW ===
# TODO: Write your solution from scratch.

File: html-css-javascript/lesson-16-js-variables-types/medium/p01-solve.js
/**
LESSON 16 — JavaScript Variables & Types
=========================================

PROBLEM: Format a Price (Medium)
Write formatPrice(amount, currency) -> "$1,234.50" style strings.

TRY THIS:
  - Map currency codes to symbols ({ USD: "$", EUR: "€", ... }).
  - Number(amount).toFixed(2), then insert thousands separators
    (regex or toLocaleString-style logic).
  - Template literal for the final string.

EXPECTED OUTPUT:
  formatPrice(1234.5, "USD") === "$1,234.50"
  formatPrice(99.999, "EUR") === "€100.00"

TEST: node medium/p01-solve.js

CHECK: python3 check.py medium/p01
*/

// TODO: Write your complete solution from scratch below.
